qiskit/transpiler/synthesis/permrowcol.py

```diff
@@ -198,3 +195,42 @@ def _eliminate_row(self, parity_mat: np.ndarray, root: int, terminals: np.ndarra
             parity_mat[edge[0], :] = (parity_mat[edge[0], :] + parity_mat[edge[1], :]) % 2
 
         return C
+
+    def _matrix_edit(
+        self, parity_mat: np.ndarray, chosen_column: int, chosen_row: int, circuit: QuantumCircuit
+    ) -> QuantumCircuit:
+        """Checks if the sum of the chosen row is bigger than one. If the sum is bigger than one, the function performs row elimination.
+
+        Args:
+            parity_mat (np.ndarray): parity matrix representing a circuit
+            chosen_column (int): index of the column to be eliminated
+            chosen_row (int): index of the row to be eliminated
+            circuit (QuantumCircuit): quantum circuit to be synthesized
+
+
+        Returns:
+            QuantumCircuit: quantum circuit with chosen row eliminated.
+
+        """
+
+        if sum(parity_mat[chosen_row]) > 1:
+
+            A = np.delete(np.delete(parity_mat.copy(), chosen_row, 0), chosen_column, 1).astype(
+                int
+            )  # Parity_mat without chosen_column and chosen_row
+            B = np.delete(parity_mat[chosen_row], chosen_column)  # Chosen_row without chosen_column
+
+            inv_A = LinearFunction(
+                LinearFunction(A).synthesize().reverse_ops()
+            ).linear  # Creates inverse of parity_mat
+
+            X = np.insert((np.matmul(B, inv_A) % 2), chosen_row, 1)  # Calculates B*inv_A
+
+            nodes = [
```

Comment:
Could we have it so that this method only returns the nodes for easier testing. Then, _eliminate_row runs after this method is called in perm_row_col